ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.codahale.metrics.Timer;
+import com.github.ambry.clustermap.DiskId;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.store.StoreMetrics.DiskLevelMetrics;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ScheduledExecutorService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Manages all the stores on a disk.
+ */
+class DiskManager {
+  private final ConcurrentMap<PartitionId, BlobStore> stores = new ConcurrentHashMap<>();
+  private final DiskId disk;
+  private final StoreConfig config;
+  private final DiskLevelMetrics metrics;
+
+  private static final Logger logger = LoggerFactory.getLogger(DiskManager.class);
+
+  /**
+   * Constructs a {@link DiskManager}
+   * @param disk representation of the disk.
+   * @param replicas all the replicas on this disk.
+   * @param config the settings for store configuration.
+   * @param scheduler the {@link ScheduledExecutorService} for executing background tasks.
+   * @param storeMetrics the {@link StoreMetrics} object used for store-related metrics.
+   * @param keyFactory the {@link StoreKeyFactory} for parsing store keys.
+   * @param recovery the {@link MessageStoreRecovery} instance to use.
+   * @param hardDelete the {@link MessageStoreHardDelete} instance to use.
+   * @param time the {@link Time} instance to use.
+   */
+  DiskManager(DiskId disk, List<ReplicaId> replicas, StoreConfig config, ScheduledExecutorService scheduler,
+      StoreMetrics storeMetrics, StoreKeyFactory keyFactory, MessageStoreRecovery recovery,
+      MessageStoreHardDelete hardDelete, Time time) {
+    this.disk = disk;
+    this.config = config;
+    this.metrics = storeMetrics.createDiskLevelMetrics(disk.getMountPath());
```

Comment:
As discussed, we found that it would be better to have aggregated metrics at the `StoreManager` level, so I will remove `DiskLevelMetrics` until we have a stronger use case for it.